src/JuliaInterpreter.jl

```diff
@@ -129,6 +129,49 @@ end
 JuliaStackFrame(frame::JuliaStackFrame, pc::JuliaProgramCounter; kwargs...) =
     JuliaStackFrame(frame.code, frame, pc; kwargs...)
 
+"""
+`Variable` is a struct representing a variable with an asigned value.
+By calling the function `locals`[@ref] on a `JuliaStackFrame`[@ref] a
+`Vector` of `Variable`'s is returned.
+
+Important fields:
+- `value::Any`: the value of the local variable
+- `name::Symbol`: the name of the variable as given in the source code
+- `isparam::Bool`: if the variable is a type parameter, for example `T` in `f(x::T) where {T} = x` .
+"""
+struct Variable
+    value::Any
+    name::Symbol
+    isparam::Bool
+end
+Base.show(io::IO, var::Variable) = print(io, var.name, " = ", var.value)
+Base.isequal(var1::Variable, var2::Variable) = 
+    var1.value == var2.value && var1.name == var2.name && var1.isparam == var2.isparam
+
+"""
+    local_variables = locals(frame::JuliaStackFrame)::Vector{Variable}
+
+Return the local variables as a vector of `Variable`[@ref].
+"""
+function locals(frame::JuliaStackFrame)
+    vars = Variable[]
+    for i = 1:length(frame.locals)
+        if !isa(frame.locals[i], Nothing)
+            # Remove #self# here in cases where it is just the method itself?
+            push!(vars, Variable(something(frame.locals[i]), frame.code.code.slotnames[i], false))
+        end
+    end
+    if frame.code.scope isa Method
+        sig = frame.code.scope.sig
+        for i in 1:length(frame.sparams)
+            @assert sig isa UnionAll 
+            push!(vars, Variable(frame.sparams[i], Symbol(sig.var.name), true))
+            sig = sig.body
+        end
+    end
+    return unique(v -> v.name, Iterators.reverse(vars))
```

Comment:
Maybe this should have a `reverse` on it?

Reply:
It occurs to me that this is what `frame.last_reference` seems to be for. But I'm not sure if these can ever "go backwards." @Keno, any insight?

Reply:
Maybe? I'm not entirely sure what's being asked though ;).

Reply:
Fair enough. This is mostly in reference to phenomena like https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/54#issuecomment-467489347. Questions:
- what's the maximum number of slots that will share the same slotname? (2?)
- do these exhibit SSA-like properties? That is, once `x@_5` gets assigned do we have to worry about `x@_2` ever holding the most recent value? In other words, do we really need that `Dict` or can we just scan `locals` backwards?

Reply:
Ah, I don't really know the details - this is an artifact of lowering, so you may have to ask @JeffBezanson

Reply:
Actually, Jeff's example inhttps://github.com/JuliaLang/julia/issues/31255#issuecomment-469823809 makes me think we definitely need the Dict (good choice in #94, @KristofferC). 

```julia
julia> function f(x)
           x = let x = x
               x = let x = x
                   x = let x = x
                       x + 1
                   end
                   x + 1
               end
               x + 1
           end
           x
       end
f (generic function with 1 method)

julia> @code_lowered f(1)
CodeInfo(
1 ─      x@_6 = x@_2
│   %2 = x@_6
│        x@_3 = %2
│   %4 = x@_3
│        x@_4 = %4
│   %6 = x@_4
│        x@_5 = %6
│        x@_4 = x@_5 + 1
│        x@_3 = x@_4 + 1
│        x@_6 = x@_3 + 1
└──      return x@_6
)
```

Reply:
Part of the problem is that in lowered IR, the scoping for variables is no longer obvious (and the mapping from surface syntax locations where it is to lowered IR where it is not).

Reply:
> the scoping for variables is no longer obvious

Yep https://github.com/JuliaDebug/JuliaInterpreter.jl/pull/94#issuecomment-469656569